rotkehlchen/chain/bitcoin/__init__.py

```diff
@@ -9,62 +7,94 @@
 from rotkehlchen.utils.network import request_get_dict
 
 
-def _have_bc1_accounts(accounts: List[BTCAddress]) -> bool:
+class BchNotSupported(Exception):
+    ...
+
+
+def _have_bc1_accounts(accounts: list[BTCAddress]) -> bool:
     return any(account.lower()[0:3] == 'bc1' for account in accounts)
 
 
-def get_bitcoin_addresses_balances(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
-    """Queries blockchain.info or blockstream for the balances of accounts
+def _query_normal_api(base_url: str, account: BTCAddress) -> FVal:
+    url = base_url + account
+    response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
+    stats = response_data['chain_stats']
+    balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
+    return satoshis_to_btc(balance)
+
+
+def _query_blockstream_info(accounts: list[BTCAddress]) -> dict[BTCAddress, FVal]:
+    base_url = 'https://blockstream.info/api/address/'
+    return {account: _query_normal_api(base_url, account) for account in accounts}
+
+
+def _query_mempool_space(accounts: list[BTCAddress]) -> dict[BTCAddress, FVal]:
+    base_url = 'https://mempool.space/api/address/'
+    return {account: _query_normal_api(base_url, account) for account in accounts}
+
+
+def _query_blockchain_info(accounts: list[BTCAddress]) -> dict[BTCAddress, FVal]:
+    if _have_bc1_accounts(accounts):
```

Comment:
Just noticed. We use explicit `is True` and `is False` checks. Pls use it here and elsewhere